app/src/main/java/org/oppia/app/hintsandsolution/HintsViewModel.kt

```diff
@@ -36,36 +36,52 @@ class HintsViewModel @Inject constructor() : HintsAndSolutionItemViewModel() {
     itemList.clear()
     for (index in hintList.indices) {
       if (itemList.isEmpty()) {
-        val hintsAndSolutionViewModel = HintsViewModel()
-        hintsAndSolutionViewModel.title.set(hintList[index].hintContent.contentId)
-        hintsAndSolutionViewModel.hintsAndSolutionSummary.set(hintList[index].hintContent.html)
-        hintsAndSolutionViewModel.isHintRevealed.set(hintList[index].hintIsRevealed)
-        itemList.add(hintsAndSolutionViewModel as HintsAndSolutionItemViewModel)
-      } else if (index == itemList.size) {
-        val isLastHintRevealed = (itemList[index - 1] as HintsViewModel).isHintRevealed.get()!!
-        if (isLastHintRevealed) {
-          val hintsAndSolutionViewModel = HintsViewModel()
-          hintsAndSolutionViewModel.title.set(hintList[index].hintContent.contentId)
-          hintsAndSolutionViewModel.hintsAndSolutionSummary.set(hintList[index].hintContent.html)
-          hintsAndSolutionViewModel.isHintRevealed.set(hintList[index].hintIsRevealed)
-          itemList.add(hintsAndSolutionViewModel as HintsAndSolutionItemViewModel)
+        addHintToList(hintList[index])
+      } else {
+        val isLastHintRevealed =
+          (itemList[itemList.size - 2] as HintsViewModel).isHintRevealed.get()!!
```

Comment:
Here and elsewhere: can we avoid the !! operator by handling the null cases?

Reply:
Done. By making this change:
`(itemList[itemList.size - 2] as HintsViewModel).isHintRevealed.get() ?: false`